### DISCUSSION TITLE
dispatchIf & dispatchUnless a closure

### DISCUSSION THREAD

**Discussion by bayukartiko** [laravel/framework#52988] [Ideas] [answered]
I have read the documentation about [`dispatchIf` and `dispatchUnless`](https://laravel.com/docs/11.x/queues#dispatching-jobs).

Are laravel has same behaviour for [`dispatching closures`](https://laravel.com/docs/11.x/queues#queueing-closures) like: 
```
  dispatchIf($condition, function () {
      // ...
  });
  dispatchUnless($condition, function () {
      // ...
  });
```
?

I think that's great feature if we also have that.

**Comment by bayukartiko** [comment, marked as answer]
for temporary solution:

```php
    $function = function () {
        // ...
    };
    $condition = true;
    
    if($condition){
        dispatch($function);
    }else{
        $function();
    }
```

**Reply by henzeb** [reply]
dispatchIf on jobs does not execute the job if false. It does not dispatch the job. This code does something else then the one you mentioned.

With the newest laravel to simulate dispatchIf for closure it is better to do

```php
when($condition, fn() => dispatch(fn() => /** your action **/ ));

unless($condition, fn() => dispatch(fn() => /** your action **/ ));
```

https://laravel.com/docs/11.x/helpers#method-when
https://laravel.com/docs/11.x/helpers#method-unless
